metals/src/main/scala/scala/meta/internal/metals/debug/SourcePathAdapter.scala

```diff
@@ -25,7 +27,7 @@ private[debug] final class SourcePathAdapter(
   private val saveJarFileToDisk = !supportVirtualDocuments
   private val dependencies = workspace.resolve(Directories.dependencies)
 
-  def toDapURI(sourcePath: AbsolutePath): Option[URI] = {
+  def toDapURI(sourcePath: AbsolutePath): Option[URI] = Try {
```

Comment:
`sourcePath` is `AbsolutePath`, so I would probably need to do `sourcePath.toNio.toString.startswith("jar:")` which is not pretty and I am not sure if it solves the issue, since it's hard to reproduce.

Also there would need to be a different condition in the next branch and then have an else that does the same as the first branch.

That would be a lot of changes for something that seems a rare exceptional scenario.

But coming to think of it it's probably better to do.

Reply:
And I now realised that we actually don't ever test breakpoint in libraries, need to work on that :/